src/machine/pwm.go

```diff
@@ -19,3 +19,19 @@ type PWMConfig struct {
 	//
 	Period uint64
 }
+
+type pwmer interface {
+	Configure(config PWMConfig) error
```

Comment:
This feels like a lot of API surface at a glance. For starters, `Enable` and `IsEnabled` might not be present in all targets since usually a call to `Configure` enables the PWM on most if not all targets. `Counter` and `SetCounter` could be removed too if it is not implemented by other targets, have not seen it used much. Is `Get` a common method in other target implementations?

Once this interface is added, methods can't be removed without breaking backward compatibility. It would be best to be conservative.

### That said...
@deadprogram @aykevl I have just had the sudden realization that users will not be able to cast `pwmer` to `*pwmGroup` since `*pwmGroup` is not exported! This would mean users would be stuck with the `pwmer` type once they call `PWMforPin` (unless they resort to `unsafe` trickery). 

### Possible solution
PWM objects are `pwmer`s but we allow users to access the underlying hardware via a call to `Hardware`. `PWMGroup` contains the target-specific methods of PWM control. 
```go
// Add a Hardware method
type pwmer interface {
    Configure()
    // ...
    Hardware() *pwmGroup // Option 0
    Hardware() *PWMGroup // Option 1
}

/
var (
    // Expose PWM0-PWM15 as interfaces
    PWM0 pwmer = (*PWMGroup)(unsafe.Pointer(rp.PWMBLOCK))
    // ...
)

func PWMforPin(pin Pin) (pwmer pwmer, err error) {
    // ...
}
```
In this case we must decide if we want option 0 (unexported underlying hardware type) or option 1 (exported). I'm really not sure which is the best one. What I am sure is that if we decide to change it again in the future it would be a breaking change.

**Final note:** The `pwmer` interface encourages users to write more portable tinygo programs. However, today we have exposed PWM0-PWM15, which are *pwmGroup. Looking back it would have been in line with the aims of this PR if these were `pwmers` instead. Maybe for tinygo V2?

Reply:
Feel like we need a Rob Pike-esque API wizard for help resolving this one.

Reply:
Sorry I haven't looked at the full design yet! But just wanted to quickly note that I very intentionally didn't put a PWM interface type in the machine package. The problem is that PWM is pretty complex and every chip has a different subset of features.